### PR TITLE
Add tools/build_cleaner.py to fix BUILD dependencies

### PR DESCRIPTION
**Pull request by achoum**
Also some initial dep fix.

### CODE REVIEW

File: temporian/test/BUILD [hunk 1]
```diff
@@ -66,16 +66,3 @@ py_test(
...
-    srcs = ["public_api_test.py"],
-    data = [
-        "//docs:reference",
-    ],
-    srcs_version = "PY3",
-    deps = [
-        # already_there/absl/testing:absltest
-        "//temporian",
-    ],
-)
```
Comment (ianspektor) [inline temporian/test/BUILD:81]: 👀

Comment (achoum) [reply temporian/test/BUILD:81]: I moved it in docs/BUILD (it makes my life a bit easier with the sync with internal version).

File: tools/build_cleaner.py [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+        "//temporian/implementation/numpy_cc/operators:operators_cc",
+    ),
+}
+
+
+# Rule prefix for third party module dependencies.
+THIRD_PARTY_RULE_PREFIX = "# already_there/"
+
+# Filename of BUILD files.
+BUILD_FILENAME = "BUILD"
```
Comment (rstz) [inline tools/build_cleaner.py:89]: BUILD.bazel is an alternative name for the build file and we actually use it for the top-level build file due to a name collision when building the pip package on non case-sensitive systems. Maybe include it here

**Review by rstz — APPROVED** [review]
Very cool!

File: tools/build_cleaner.py [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+This program does not modify BUILD files. Instead, it lists the dependencies to
+add and remove.
+
+How to use the build cleaner.
+
+    # Run the following command.
+    tools/build_cleaner.py
+
+If case of a false positive, you might have to update the configuration fields
+BUILD_IN_MODULES, THIRD_PARTY_MODULES, or EXTRA_SOURCE_TO_RULE.
```
[the remark this answers is not in the document]
Comment (achoum) [reply tools/build_cleaner.py:12]: *Done

File: tools/build_cleaner.py (hunk 2 above)
Comment (achoum) [reply tools/build_cleaner.py]: Good point.
*Done

**Pull request merged**